src/jarabe/desktop/favoriteslayout.py

```diff
@@ -172,7 +172,15 @@ def allocate_children(self, allocation, children):
         for child in children:
             if not self._grid.is_in_grid(child):
                 width, height = self._get_child_grid_size(child)
-                self._grid.add(child, width, height, None, None, locked=False)
+                x = y = None
+                if hasattr(child, "_buddy"):
```

Comment:
You typically don't want to access _ methods or props outside the class.  Maybe add a function "get_placement_key" and add that to the child class?